src/3d/qgs3dmapsettings.h

```diff
@@ -57,6 +57,12 @@ class _3D_EXPORT Qgs3DMapSettings : public QObject, public QgsTemporalRangeObjec
 {
     Q_OBJECT
   public:
+    enum ViewSyncMode
+    {
+      NoSync, //! No syncronisation will happen
+      Sync3DTo2D, //! Syncronize 3D view camera to the main map canvas extent
```

Comment:
are these options mutually exclusive? i don't see why we couldn't have full two way sync (if the user opts for this). i.e. move camera in 3d view = 2d map moves, move map in 2d view = camera moves in 3d.

IMO the NoSync option should be removed, and these should become flags instead of exclusive options.